### ISSUE TITLE
VeraCrypt will not compile NOGUI on Linux

### ISSUE THREAD

**Issue by juanitotc** [veracrypt/VeraCrypt#221]
Why does make look for wxWidgets in the NOGUI case?

$ make NOGUI=1
...
Compiling CoreServiceResponse.cpp
Compiling CoreUnix.cpp
Compiling CoreLinux.cpp
Updating library Core.a
make[1]: wx-config: Command not found
make[1]: wx-config: Command not found
Precompiling SystemPrecompiled.h
SystemPrecompiled.h:13:19: fatal error: wx/wx.h: No such file or directory
 #include <wx/wx.h>
                   ^
compilation terminated.
make[1]: *** [/usr/src/VeraCrypt/src/Build/Include/Makefile.inc:46: SystemPrecompiled.h.gch] Error 1
make: *** [Makefile:369: all] Error 2
Command exited with non-zero status 2

**Comment by gv5470** [comment]
Hello @juanitotc,
wxWidgets embeds many functionalities, most of them being GUI-related, but also other such as files/ memory access or streams management, all of it in a portable fashion. If you look at the headers included by SystemPrecompiled.h:
```#include <wx/wx.h>
#include <wx/filename.h>
#include <wx/mstream.h>
#include <wx/snglinst.h>
#include <wx/txtstrm.h>
#include <wx/wfstream.h>
```

you will see that even the no-gui version uses wxWidgets for that kind of functionalities (thus making the code more portable and OS-agnostic, which is presumably the main point).

**Comment by kkrevenant** [comment]
Just telling you a true story: I was compiling this on Gentoo musl, the wxWigets dependency trickled down and eventually pulled in some desktop icon set called Adwaita, which failed because some image converter cannot load png images. I eventually managed to get around that and had it installed, but for a guy who never intend to use veracrypt in GUI mode ever - I was doing it on a headless docker host, that experience was distasteful at the best ....

**Comment by danieldjewell** [comment]
@juanitotc and @kkrevenant 

I'm not 100% sure that it would fix the problem, but have you tried building wxWidgets first and then building the nogui? It appears that using static linking for the wxWidgets resolves the problem... 

https://github.com/veracrypt/VeraCrypt/blob/master/src/Readme.txt#L171

You can clone the source from https://www.github.com/wxWidgets/wxWidgets

I recommend doing a checkout of a release tag if you're going to git clone (instead of downloading a source archive) ... for example: 

```
git checkout tags/v3.0.4 -b v3.0.4
``` 

I was able to get this working with v3.0.4 of wxWidgets and the latest master version as of today of veracrypt (with the exception of an edit to fix #427 on GCC).

**Comment by juanitotc** [comment]
Building wxwidgets first and then compiling veracrypt --nogui works, but it seems a waste to required wxwidgets and all of its dependencies when no gui is used..
